src/domain/model/user/User.ts

```diff
@@ -47,4 +58,27 @@ export class User {
   get modificationTime() {
     return this._modificationTime;
   }
+
+  assignRole(role: Role, actor: UserId) {
+    this.roleAssignments.addAssignment(role, this, actor);
+  }
+
+  removeRole(role: Role, actor: UserId) {
+    this.roleAssignments.removeAssignment(role, this, actor);
+  }
+
+  get roles(): string[] {
+    return this.roleAssignments.activeAssignments().map((role) => role.name);
+  }
+
+  get permissions(): string[] {
+    const permissions: Set<string> = new Set();
+    this.roleAssignments.activeAssignments().forEach((role) => {
+      role
+        .permissions()
+        .map((permission) => permission.name)
+        .forEach((permissionName) => permissions.add(permissionName));
```

Comment:
This shouldn't need the parameter brackets, surprised the formatter isn't removing them?

Reply:
Which brackets do you mean? Around `permissionName`? They are always added by the formatter even for single parameter lambdas

Reply:
Ok, weird, but fine.